tests/utils/graph_generator.py

```diff
@@ -0,0 +1,320 @@
+
+        ```python
+        gen = GraphGenerator()
+
+        # Pre-create a battery component to refer to it later or to use it in multiple
+        # places in the graph.
+        special_bat = gen.component(ComponentCategory.BATTERY)
+
+        graph = gen.to_graph(
+            (
+                ComponentCategory.METER, # grid side meter
+                [ # list of components connected to grid side meter
+                    (
+                        ComponentCategory.METER, # Meter in front of battery->inverter
+                        ( # A tuple, first is connected to parent, second is the child of the first
+                            # Inverter in front of battery, type will be
+                            # set to InverterType.BATTERY
+                            ComponentCategory.INVERTER,
+                            ComponentCategory.BATTERY,  # Battery
+                        ),
+                    ),
+                    (
+                        ComponentCategory.METER,
+                        (
+                            # Inverter in front of battery, type will be
+                            # set to InverterType.BATTERY
+                            ComponentCategory.INVERTER,
+                            special_bat,  # Pre-created battery
+                        ),
+                    ),
+                ],
+            )
+        )
+        ```
+
+        Args:
+            components: the components to convert to a graph.
+
+        Returns:
+            a tuple containing the components and connections of the graph.
+        """
+        graph = self._to_graph((self.grid(), components))
+        return _MicrogridComponentGraph(set(graph[0]), set(graph[1]))
+
+    def _to_graph(
+        self, components: tuple[Any, Any]
+    ) -> tuple[list[Component], list[Connection]]:
+        """Convert a list of components to a graph.
+
+        Args:
+            components: the components to convert to a graph.
+
+        Returns:
+            a tuple containing the components and connections of the graph.
+        """
+        component: Component
+
+        # Handle both Component and ComponentCategory types
+        if isinstance(components[0], Component):
+            component = components[0]
+        elif isinstance(components[0], ComponentCategory):
+            component = self.component(components[0])
+        else:
+            raise ValueError("Invalid component list")
+
+        assert isinstance(component, Component)
+        assert component is not None
+
+        def inverter_type(category: ComponentCategory) -> InverterType | None:
+            match category:
+                case ComponentCategory.BATTERY:
+                    return InverterType.BATTERY
+                case ComponentCategory.CHP:
+                    return InverterType.SOLAR
+                case ComponentCategory.EV_CHARGER:
+                    return InverterType.HYBRID
+                case _:
+                    return None
```

Comment:
CHP and EV Chargers don't have inverters in front of them.  And we don't have a PV component type,  so we can guess this only if there's a Battery as a successor.  Actually even then not 100%,  because it could be a HYBRID inverter (which handle batteries and PV in the same inverter).  But we can default to Battery inverter in this case.

And for the other types of inverters, we can say they need to created manually, with the `gen.component()` method that you're using.

```suggestion
        def inverter_type(successor_category: ComponentCategory) -> InverterType | None:
            match successor_category:
                case ComponentCategory.BATTERY:
                    return InverterType.BATTERY
                case _:
                    raise TypeError(f"Found an inverter in front of a {successor_category}")
```

Reply:
> And for the other types of inverters, we can say they need to created manually, with the gen.component() method that you're using.

Right, but it's also valid to have a `None` type for the inverter, right?

Reply:
I think people can use `InverterType.NONE` instead.